### ISSUE TITLE
The version for log4cplus "1.1.3-rc6" is not parsed correctly

### ISSUE THREAD

**Issue by thozza** [rebase-helper/rebase-helper#144]
I'm creating this report so that we don't forget about it.
The version and pre-version extraction for log4cplus does not work correctly. The code for extracting the version could be improved to better reuse the SOURCE0 string for extraction regexp.

Output of rebase-helper:
```
thozza@thozza-pc ~/devel/fedora/log4cplus (master %>)
$ rebase-helper 1.1.3-rc6
Git-rebase operation to log4cplus-1.1.3-rc3 is ongoing...
Building packages for log4cplus version 1.1.3
Building SRPM
Building SRPM finished successfully
Building RPMs
^C
Interrupted by user
```

You can see that the "rc6" is not extracted. Also it may be worth indicating that basically part of the passed version was ignored and it may not make sense to continue with the rebase.

Output in verbose mode:
```
thozza@thozza-pc ~/devel/fedora/log4cplus (master %>)
$ rebase-helper -v 1.1.3-rc6
Results directory 'rebase-helper-results' exists, removing it
Updating the extra version
Extracting version from 'log4cplus-1.1.3-rc3.tar.xz' using '%{name}-%{version}%{?prever:-%{prever}}.tar.xz'
Extracting version using regex '%{name}-([.0-9]+\w*)%{?prever:-%{prever}}.tar.xz'
Failed to extract version from archive name!
Trying to extracting version using fallback regex '^\w+-?_?v?([.0-9]+\w*)(.tgz|.tar.gz|.gem|.bz2|.tar.xz|.zip)'
Failed to extract version from archive name using fallback regex!
Splitting string ''
Updating the extra version
Extracting version from 'log4cplus-1.1.3-rc3.tar.xz' using '%{name}-%{version}%{?prever:-%{prever}}.tar.xz'
Extracting version using regex '%{name}-([.0-9]+\w*)%{?prever:-%{prever}}.tar.xz'
Failed to extract version from archive name!
Trying to extracting version using fallback regex '^\w+-?_?v?([.0-9]+\w*)(.tgz|.tar.gz|.gem|.bz2|.tar.xz|.zip)'
Failed to extract version from archive name using fallback regex!
Splitting string ''
argument passed as a new source is a version
Splitting string '1.1.3-rc6'
Divided version '1.1.3' and extra string ''
Updating version in SPEC from '1.1.3' with '1.1.3'
Writing SPEC file '/home/thozza/devel/fedora/log4cplus/rebase-helper-results/log4cplus.spec' to the disc
Updating the extra version
Extracting version from 'log4cplus-1.1.3-rc3.tar.xz' using '%{name}-%{version}%{?prever:-%{prever}}.tar.xz'
Extracting version using regex '%{name}-([.0-9]+\w*)%{?prever:-%{prever}}.tar.xz'
Failed to extract version from archive name!
Trying to extracting version using fallback regex '^\w+-?_?v?([.0-9]+\w*)(.tgz|.tar.gz|.gem|.bz2|.tar.xz|.zip)'
Failed to extract version from archive name using fallback regex!
Splitting string ''
Updating extra version in SPEC to ''
Changing release line to 'Release: 1%{?prever:.%{prever}}%{?dist}'
Writing SPEC file '/home/thozza/devel/fedora/log4cplus/rebase-helper-results/log4cplus.spec' to the disc
Updating the extra version
Extracting version from 'log4cplus-1.1.3-rc3.tar.xz' using '%{name}-%{version}%{?prever:-%{prever}}.tar.xz'
Extracting version using regex '%{name}-([.0-9]+\w*)%{?prever:-%{prever}}.tar.xz'
Failed to extract version from archive name!
Trying to extracting version using fallback regex '^\w+-?_?v?([.0-9]+\w*)(.tgz|.tar.gz|.gem|.bz2|.tar.xz|.zip)'
Failed to extract version from archive name using fallback regex!
Splitting string ''
Writing SPEC file '/home/thozza/devel/fedora/log4cplus/rebase-helper-results/log4cplus.spec' to the disc
Updating the extra version
Extracting version from 'log4cplus-1.1.3-rc3.tar.xz' using '%{name}-%{version}%{?prever:-%{prever}}.tar.xz'
Extracting version using regex '%{name}-([.0-9]+\w*)%{?prever:-%{prever}}.tar.xz'
Failed to extract version from archive name!
Trying to extracting version using fallback regex '^\w+-?_?v?([.0-9]+\w*)(.tgz|.tar.gz|.gem|.bz2|.tar.xz|.zip)'
Failed to extract version from archive name using fallback regex!
Splitting string ''
Workspace directory 'rebase-helper-workspace' exists, removing it
...
```

rebase-helper version used:
```
$ rpm -q rebase-helper
rebase-helper-0.6.2-1.fc22.noarch
```

Feel free to leave the issue for me, since I wrote the code for parsing the version string.

**Comment by phracek** [comment]
I have discovered the same issue with python-docker-py.

BZ https://bugzilla.redhat.com/show_bug.cgi?id=1320761#c7
Case: python-docker-py-1.8.0rc5 goes to %{projec%{REBASE_VER} which is wrong.
It seems, rebase-helper substitutes versions wrongly.
~~~
phracek@unused-4-101: ~/work/Fedora/python-docker-py (master %)$ cat rebase-helper-results/python-docker-py.spec
%global REBASE_EXTRA_VER rc5
%global REBASE_VER %{version}%{REBASE_EXTRA_VER}
%if 0%{?fedora}
%global         with_python3    1
%else %if 0%{?centos}
%global         with_python3    0
%endif #fedora

# disable debug package cause archful
%global         debug_package   %{nil}

%global owner docker
%global project docker-py
%global commit cc399d22b4f0631e5901019b005aa5109dc635a0
%global shortcommit %(c=%{commit}; echo ${c:0:7})

Name:           python-docker-py
Version:        1.8.0
#Release:        0.1%{?dist}
Release:        0.1%{?REBASE_EXTRA_VER}%{?dist}
Summary:        An API client for docker written in Python
License:        ASL 2.0
URL:            https://github.com/%{owner}/%{project}/
#Source0:        https://github.com/%{owner}/%{project}/archive/%{commit}/%{project}-%{commit}.tar.gz
Source0:        https://github.com/%{owner}/%{project}/archive/%{commit}/%{projec%{REBASE_VER}.tar.gz
~~~

**Comment by phracek** [comment]
Fixed by #215
